app/services/cached_commodity_service.rb

```diff
@@ -0,0 +1,148 @@
+      measure_components: [
+        { duty_expression: :duty_expression_description },
+        { measurement_unit: %i[measurement_unit_description measurement_unit_abbreviations] },
+        { measure: { measure_type: :measure_type_description } },
+        :monetary_unit,
+        :measurement_unit_qualifier,
+      ],
+    },
+    {
+      measure_conditions: [
+        { measure_action: :measure_action_description },
+        { certificate: :certificate_descriptions },
+        { certificate_type: :certificate_type_description },
+        { measurement_unit: %i[measurement_unit_description measurement_unit_abbreviations] },
+        :monetary_unit,
+        :measurement_unit_qualifier,
+        { measure_condition_code: :measure_condition_code_description },
+        {
+          measure_condition_components: [
+            { measurement_unit: %i[measurement_unit_description measurement_unit_abbreviations] },
+            :measure_condition,
+            :duty_expression,
+            :monetary_unit,
+            :measurement_unit_qualifier,
+          ],
+        },
+      ],
+    },
+    { quota_order_number: { quota_definition: %i[quota_balance_events quota_suspension_periods quota_blocking_periods] } },
+    { excluded_geographical_areas: :geographical_area_descriptions },
+    { geographical_area: [:geographical_area_descriptions,
+                          { contained_geographical_areas: :geographical_area_descriptions }] },
+    { additional_code: :additional_code_descriptions },
+    :footnotes,
+    :base_regulation,
+    :modification_regulation,
+    :full_temporary_stop_regulations,
+    :measure_partial_temporary_stops,
+  ].freeze
+
+  TTL = 24.hours
+
+  def initialize(commodity, actual_date, filter_params)
+    @commodity = commodity
+    @actual_date = actual_date
+    @filter_params = filter_params
+  end
+
+  def call
+    Rails.cache.fetch(cache_key, expires_in: TTL) do
+      Api::V2::Commodities::CommoditySerializer.new(presented_commodity, options).serializable_hash
+    end
+  end
+
+  private
+
+  attr_reader :commodity, :actual_date, :filter_params
+
+  def presented_commodity
+    Api::V2::Commodities::CommodityPresenter.new(commodity, presented_measures)
+  end
+
+  def presented_measures
+    # TODO: This should be Api::V2::Measures::MeasurePresenter. It works currently because the real presenter used by the CommodityPresenter uses the correct V2 measures presenter.
+    #       Also, why are we validating in a presenter?
+    MeasurePresenter.new(measures, commodity).validate!
+  end
+
+  def options
+    {
+      is_collection: false,
+      include: DEFAULT_COMMODITY_INCLUDES,
+    }
+  end
+
+  def measures
+    measures = commodity.measures_dataset.eager(*MEASURES_EAGER_LOAD_GRAPH).all
+    if filter_by_country_id? && filtering_country.present?
+      measures.select do |measure|
+        measure.relevant_for_country?(filtering_country.geographical_area_id)
+      end
+    else
+      measures
+    end
```

Comment:
This could be in a separate method maybe?
```ruby
def apply_filter(measures, filtering_country)
return measures unless filtering_country.present? || filter_by_country_id?

measures.select do |measure|
    measure.relevant_for_country?(filtering_country.geographical_area_id)
  end
end 
```